base/shared/src/main/scala/scalaz/data/Kleisli.scala

```diff
@@ -0,0 +1,35 @@
+package scalaz
+package data
+
+sealed trait KleisliModule {
+  type Kleisli[F[_], A, B]
+
+  def runKleisli[F[_], A, B](k: Kleisli[F, A, B]): A => F[B]
+
+  def wrapKleisli[F[_], A, B](k: A => F[B]): Kleisli[F, A, B]
+
+  def hoist[F[_], G[_], A, B](k: Kleisli[F, A, B])(η: F ~> G): Kleisli[G, A, B]
+
+  def compose[F[_], A, B, C](
+    j: Kleisli[F, B, C],
+    k: Kleisli[F, A, B]
+  )(implicit M: Monad[F]): Kleisli[F, A, C]
+
+}
+
+private[data] object KleisliImpl extends KleisliModule {
+  type Kleisli[F[_], A, B] = A => F[B]
+
+  override def runKleisli[F[_], A, B](k: Kleisli[F, A, B]): A => F[B] = k
+
+  override def wrapKleisli[F[_], A, B](k: A => F[B]): Kleisli[F, A, B] = k
+
+  override def hoist[F[_], G[_], A, B](k: Kleisli[F, A, B])(η: F ~> G): Kleisli[G, A, B] =
+    k andThen η.apply
+
+  override def compose[F[_], A, B, C](
+    j: Kleisli[F, B, C],
+    k: Kleisli[F, A, B]
+  )(implicit M: Monad[F]): Kleisli[F, A, C] =
```

Comment:
Can this be relaxed to a `Bind` constraint? Will have ripples down for all consumers if so.

Reply:
ooh, good catch I think

Reply:
It can in most places, yes. I went for homogeneity because the `&&&` requires `pure` and `flatMap`

Reply:
I think the principle of requiring the least powerful constraint is important. Does the stronger constraint on `&&&` cause other issues?

Reply:
Yes, because apparently an implicit `Monad[F]` doesn't expose an implicit `Bind[F]` - check the gitter - I've posted screenshots